### PR TITLE
Add vexctl create subcommand

### PR DESCRIPTION
**Pull request by puerco**
This PR adds a new subcommand to vexctl to create simple one statement documents. The idea of this subcommand is to handle simple use cases and to help folks to get started with a minimal document.  This is intended to address some points raised in https://github.com/chainguard-dev/vex/issues/38

This PR also introduces a canonicalization algorithm in f3f03f3546329d09dcd930b5278a3fdf12ef002a to generate a hash identifying the impact data in documents. The idea is to be able to generate a constant hash derived from the VEX information in the document that does not change if superfluous information does.

**From the help text:**

```
vexctl create: creates a new VEX document

The create subcommand generates a single statement document
from the command line. This is intended for simple use cases
or to get a base document to get started.

You can specify multiple products and customize the metadata of
the document via the command line flags. vexctl will honor the
SOURCE_DATE_EPOCH environment variable and use that date for 
the document (it can be formated in unix time or RFC3339).

If you don't specify an ID for the document, one will be generated
using its canonicalization hash.

Examples:

# Generate a document stating that CVE-2023-12345 was fixed in the 
# git package of Wolfi:

vexctl create "pkg:apk/wolfi/git@2.39.0-r1?arch=x86_64" CVE-2023-12345 fixed

# You can specify more than one product. vexctl will read one from
# the argument but you can control all parameters through command line
# flags. Here's an example with two products in the same document:

vexctl create --product="pkg:apk/wolfi/git@2.39.0-r1?arch=x86_64" \
              --product="pkg:apk/wolfi/git@2.39.0-r1?arch=armv7" \
              --vuln="CVE-2023-12345" \
              --status="fixed"

# not_affected statements need a justification:

vexctl create --product="pkg:apk/wolfi/trivy@0.36.1-r0?arch=x86_64" \
              --vuln="CVE-2023-12345" \
              --status="not_affected" \
              --justification="component_not_present"

Usage:
  vexctl create [flags] [product_id [vuln_id [status]]]

```

### CODE REVIEW

File: internal/cmd/create.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+		Long: fmt.Sprintf(`%s create: creates a new VEX document
+
+The create subcommand generates a single statement document
+from the command line. This is intended for simple use cases
+or to get a base document to get started.
+
+You can specify multiple products and customize the metadata of
+the document via the command line flags. vexctl will honor the
+SOURCE_DATE_EPOCH environment variable and use that date for 
+the document (it can be formated in unix time or RFC3339).
```
Comment (luhring) [inline internal/cmd/create.go:89]:
```suggestion
the document (it can be formatted in unix time or RFC3339).
```

File: internal/cmd/create.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	createCmd := &cobra.Command{
+		Short: fmt.Sprintf("%s create: creates a new VEX document", appname),
+		Long: fmt.Sprintf(`%s create: creates a new VEX document
+
+The create subcommand generates a single statement document
+from the command line. This is intended for simple use cases
+or to get a base document to get started.
+
+You can specify multiple products and customize the metadata of
+the document via the command line flags. vexctl will honor the
```
Comment (luhring) [inline internal/cmd/create.go:87]: It looks like you're using `%s` for the app name, did you want to use that for the "vexctl" on this line, too?

File: internal/cmd/create.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+			if _, err := newDoc.GenerateCanonicalID(); err != nil {
+				return fmt.Errorf("generating document id: %w", err)
+			}
+
+			out := os.Stdout
+
+			if opts.outFilePath != "" {
+				f, err := os.Create(opts.outFilePath)
+				if err != nil {
+					return fmt.Errorf("opening vex file to write document: %w", err)
```
Comment (luhring) [inline internal/cmd/create.go:159]: nit: I noticed sometimes we say "vex" and sometimes we say "VEX". Does that matter?

File: internal/cmd/create.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+		[]string{},
+		"list of products to list in the statement, at least one is required",
+	)
+
+	createCmd.PersistentFlags().StringVarP(
+		&opts.Status,
+		"status",
+		"s",
+		"",
+		fmt.Sprintf("status of the product vs the vulnerability, see %s show statuses for list", appname),
```
Comment (luhring) [inline internal/cmd/create.go:218]:
```suggestion
		fmt.Sprintf("status of the product vs the vulnerability, see `%s show statuses` for list", appname),
```

Maybe not this suggestion exactly, but somehow to make it clear which words are part of the command we're suggesting they run...

File: internal/cmd/create.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+		[]string{},
+		"list of subcomponents to add to the statement",
+	)
+
+	createCmd.PersistentFlags().StringVarP(
+		&opts.Justification,
+		"justification",
+		"j",
+		"",
+		fmt.Sprintf("justification for not_affected status, see %s show justifications for list", appname),
```
Comment (luhring) [inline internal/cmd/create.go:233]:
```suggestion
		fmt.Sprintf("justification for not_affected status, see `%s show justifications` for list", appname),
```

Maybe not this suggestion exactly, but somehow to make it clear which words are part of the command we're suggesting they run...

**Review by luhring — APPROVED** [review]

File: internal/cmd/create.go [hunk 6]
```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2022 Chainguard, Inc.
```
[the remark this answers is not in the document]
Comment (puerco) [reply internal/cmd/create.go:2]: You're right

File: internal/cmd/create.go (hunk 3 above)
Comment (puerco) [reply internal/cmd/create.go:159]: I've changed them all to "VEX"

File: internal/cmd/create.go (hunk 5 above)
Comment (puerco) [reply internal/cmd/create.go:233]: Sounds good enough to me, i've added the backticks

**Pull request merged**